web/handlers/http.go

```diff
@@ -328,11 +333,26 @@ func New(
 
 	mainMux.Handle("/", m)
 
+	urlTo := web.NewURLTo(m)
+	consumeURL := urlTo(router.CompleteInviteConsume)
+
 	// apply HTTP middleware
 	middlewares := []func(http.Handler) http.Handler{
 		logging.InjectHandler(logger),
 		members.ContextInjecter(dbs.Members, authWithPassword, authWithSSB),
 		CSRF,
+
+		// this is an ugly hack to disable CSRF for certain requests that are done by apps
```

Comment:
I'm unhappy with this one but I'm not 100% sure if we _really_ need CSRF on this route since the token/inviteCode is a secret somehow already.

Reply:
The alternative would be to set a header on the `/join` request (`X-CSRF-TOKEN`) and have Apps send it back but that would need another change of the spec.

(See also: https://github.com/gorilla/csrf#javascript-applications)

Reply:
If its like an API where you might have special authentication anyway then it should be fine to disable CSRF. Django has the same kind of escape hatch: https://docs.djangoproject.com/en/3.1/ref/csrf/#django.views.decorators.csrf.csrf_exempt